qcodes/instrument_drivers/Minicircuits/USB_SPDT.py

```diff
@@ -0,0 +1,88 @@
+    def __init__(self, parent, name, channel_letter):
+        """
+        Args:
+            parent (Instrument): The instrument the channel is a part of
+            name (str): the name of the channel
+            channel_letter (str): channel letter ['a', 'b', 'c' or 'd'])
+        """
+
+        super().__init__(parent, name)
+        self.channel_letter = channel_letter.upper()
+        _chanlist = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']
+        self.channel_number = _chanlist.index(channel_letter)
+
+        self.add_parameter('switch',
+                           label='switch',
+                           set_cmd=self._set_switch,
+                           get_cmd=self._get_switch,
+                           vals=vals.Ints(1, 2)
+                           )
+
+    def _set_switch(self, switch): 
+        self._parent.switch.Set_Switch(self.channel_letter,switch-1)
+
+    def _get_switch(self):
+        status = self._parent.switch.GetSwitchesStatus(self._parent.address)[1]
+        return int("{0:04b}".format(status[-1-self.channel_number]))+1
+
+
+class USB_SPDT(Instrument):
+    """
+    Mini-Circuits SPDT RF switch
+
+    Args:
+            name (str): the name of the instrument
+            address (int, optional):
+            kwargs (dict): kwargs to be passed to Instrument class.
+    """
+    def __init__(self, name, address=None, **kwargs):
+        super().__init__(name, **kwargs)
+
+        try:
+            clr.AddReference('qcodes//instrument_drivers//Minicircuits//mcl_RF_Switch_Controller64')
+        except ImportError:
+            raise ImportError("""Load of mcl_RF_Switch_Controller64.dll not possible. Make sure
+                                the dll file is not blocked by Windows. To unblock right-click 
+                                the dll to open proporties and check the 'unblock' checkmark 
+                                in the bottom. Check that your python installation is 64bit.""")
+        import mcl_RF_Switch_Controller64
+        self.switch = mcl_RF_Switch_Controller64.USB_RF_SwitchBox()
+
+        if address == None:
+            self.switch.Connect()
+            self.address = self.switch.Get_Address()
+        else:
+            self.switch.ConnectByAddress(address)
+            self.address = address
+        self.connect_message()
+
+        channels = ChannelList(self, "Channels", MC_channel,
+                               snapshotable=False)
+
+        _chanlist = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h']
+        _max_channel_number = int(self.IDN()['model'][3])
+        _chanlist = _chanlist[0:_max_channel_number]
+
+        for c in _chanlist:
+            channel = SwitchChannelUSB(self, 'channel_{}'.format(c), c)
+            channels.append(channel)
+            self.add_submodule('channel_{}'.format(c), channel)
+        channels.lock()
+        self.add_submodule('channels', channels)
+
+    def get_idn(self):
+        fw = self.switch.GetFirmware()
```

Comment:
Both commands exists in the dll but throws an error. I think they need some kind of argument but not sure...
Read_ModelName()
Read_SN()

Reply:
Just figured out the right syntax. The address has to passed as a string:
Read_ModelName(str(self.address))
Read_SN(str(self.address))

Reply:
is the idn now working with these modifications?